velox/tpch/gen/TpchGen.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/memory/Memory.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::tpch {
+
+/// This file uses TPC-H DBGEN to generate data encoded using Velox Vectors.
+///
+/// The basic input for the API is the TPC-H table name (the Table enum), the
+/// TPC-H scale factor, the maximum batch size, and the offset. The common usage
+/// is to make successive calls to this API advancing the offset parameter,
+/// until all records were read. Clients might also assign different slices of
+/// the range "[0, getRowCount(Table, scaleFactor)[" to different threads in
+/// order to generate datasets in parallel.
+///
+/// If not enough records are available given a particular scale factor and
+/// offset, less than maxRows records might be returned.
+///
+/// Data is always returned in a RowVector.
+
+enum class Table {
+  TBL_PART,
+  TBL_SUPPLIER,
+  TBL_PARTSUP,
+  TBL_CUSTOMER,
+  TBL_ORDERS,
+  TBL_LINEITEM,
+  TBL_NATION,
+  TBL_REGION,
+};
+
+/// Returns the row count for a particular TPC-H table given a scale factor, as
+/// defined in the spec available at:
+///
+///  https://www.tpc.org/tpch/
+constexpr size_t getRowCount(Table table, size_t scaleFactor);
+
+/// Returns a row vector containing at most `maxRows` rows of the "nation"
+/// table, starting at `offset`, and given the scale factor. The row vector
+/// returned has the following schema:
+///
+///  n_nationkey: BIGINT
+///  n_name: VARCHAR
+///  n_regionkey: BIGINT
+///  n_comment: VARCHAR
+///
+RowVectorPtr genTpchNation(
```

Comment:
@pedroerp It'll be nice to add an overload function with just scaleFactor and pool. It'll be the most common senarios for most use cases, and the users don't need to care about what maxRows and offset are.
```
RowVectorPtr genTpchNation(
    size_t scaleFactor,
    memory::MemoryPool* pool) {
   return genTpchNation(getRowCount(TBL_NATION, scaleFactor), 0, scaleFactor, pool);
}
```

Reply:
You mean only for Nation (because it's tiny) or for all other tables as well? But good point, I'll add.

Reply:
@pedroerp I mean for all tables. When a user wants to generate TPCH tables, he/she only needs to care about the scalefactor.

Reply:
For the larger tables we can't materialize the full dataset in a single rowVector, so the API will have to expose knobs to let users control how the splits are organized, the sizes, offsets, and such.